src/vm/core.zig

```diff
@@ -1096,6 +1136,254 @@ test "trace is disabled" {
     }
 }
 
+test "deduceOp1 when opcode == .Call" {
+    // ************************************************************
+    // *                 SETUP TEST CONTEXT                       *
+    // ************************************************************
+    var allocator = std.testing.allocator;
+    var vm = try CairoVM.init(allocator, .{});
+    defer vm.deinit();
+
+    // ************************************************************
+    // *                      TEST BODY                           *
+    // ************************************************************
+    const instruction = instructions.Instruction{
+        .off_0 = 1,
+        .off_1 = 2,
+        .off_2 = 3,
+        .dst_reg = .FP,
+        .op_0_reg = .AP,
+        .op_1_addr = .AP,
+        .res_logic = .Add,
+        .pc_update = .Jump,
+        .ap_update = .Regular,
+        .fp_update = .Regular,
+        .opcode = .Call,
+    };
```

Comment:
We are creating a new `Instruction` with all its key-value pairs for each test. We can also bring this up one level and change the values we are interested in for each test.

Something like:

```zig
var test_instruction = Instruction { ... };

test "deduceOp1 ..." {
    test_instruction.opcode = .Call;

    // rest of test
}
```

Reply:
duplicate?

but yeah agreed, I'll fix that